app/src/main/java/ai/elimu/analytics/util/VersionHelper.kt

```diff
@@ -124,6 +125,20 @@ object VersionHelper {
                 }
             }
 
+            if (oldVersionCode < 3003002) {
+                Timber.w("oldVersionCode < 3003002")
+                // Delete old folder structure
+                val filesDir = context.filesDir
+                for (file in filesDir.listFiles()) {
```

Comment:
`listFiles()` could return null, resulting in `NPE`.
```suggestion
                for (file in filesDir.listFiles() ?: emptyArray()) {
```

Reply:
Makes sense.